packages/create-block/lib/scaffold.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * External dependencies
+ */
+const { dirname, join } = require( 'path' );
+const makeDir = require( 'make-dir' );
+const { readFile, writeFile } = require( 'fs' ).promises;
+const { render } = require( 'mustache' );
+
+/**
+ * Internal dependencies
+ */
+const initWPScripts = require( './init-wp-scripts' );
+const { code, clear, info, success } = require( './log' );
+const { hasWPScriptsEnabled, getOutputFiles } = require( './templates' );
+
+module.exports = async function( templateName, {
+	namespace,
+	slug,
+	title,
+	description,
+	dashicon,
+	category,
+	author,
+	license,
+	version,
+} ) {
+	clear();
+	info( `Creating a new WordPress block in "${ slug }" folder.` );
+
+	const outputFiles = getOutputFiles( templateName );
+	const view = {
+		namespace,
+		slug,
+		machineName: `${ namespace }_${ slug }`.replace( /\-/g, '_' ),
+		title,
+		description,
+		dashicon,
+		category,
+		version,
+		author,
+		license,
+		textdomain: namespace,
+	};
+
+	await Promise.all(
+		Object.keys( outputFiles ).map( async ( fileName ) => {
+			const template = await readFile(
+				join( __dirname, `templates/${ outputFiles[ fileName ] }.mustache` ),
+				'utf8'
+			);
+			const filePath = `${ slug }/${ fileName.replace( /\$slug/g, slug ) }`;
+			await makeDir( dirname( filePath ) );
+			writeFile(
+				filePath,
+				render( template, view )
+			);
+		} )
+	);
+
+	if ( hasWPScriptsEnabled( templateName ) ) {
+		await initWPScripts( view );
+	}
+
+	clear();
```

Comment:
For usability, I'm not sure I like how we clear here, or really: how we clear at all. I let the scaffolding script run in the background, and when I returned, I was a bit confused because I wanted to see the history of the preceding commands. But the history was erased. It feels a bit more obtrusive than I think we need to be.

Reply:
Let's remove it then. I followed `create-react-app` 😅